sdk/servicebus/azure-servicebus/api_samples.py

```diff
@@ -0,0 +1,219 @@
+    session_id="test_session"
+)
+
+# Topic Sender
+topic_sender = ServiceBusSender.from_connection_string(
+    conn_str="conn_str",
+    topic_name="topic_name"
+)
+
+topic_sender = ServiceBusSender(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+# Subscription Receiver
+subscription_receiver = ServiceBusReceiver.from_connection_string(
+    conn_str="conn_str",
+    topic_name="topic_name",
+    subscription_name="subscription_name"
+)
+
+subscription_receiver = ServiceBusReceiver(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    subscription_name="subcription_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session_id="test_session"
+)
+
+
+### Send and receive
+# 1 Send
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+queue_sender = sb_client.get_queue_sender(queue_name="queue_name")
+
+with sb_client:
+    with queue_sender:
+        # 1.1 send single
+        message = Message("Test message")
+        queue_sender.send(message)
+
+        # 1.2 send batch
+        batch_message = topic_sender.create_batch(max_size_in_bytes=256 * 1204)
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        topic_sender.send(batch_message)
+
+        # 1.3 schedule
+        schedule_message = Message("scheduled_message")
+
+        schedule_time_utc = datetime.utcnow() + timedelta(minutes=10)
+        sequence_number = topic_sender.schedule(schedule_message, schedule_time_utc)
+
+        batch_schedule_message = topic_sender.create_batch(max_size_in_bytes=256 * 1204)
+
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        batch_sequence_numbers = topic_sender.schedule(batch_schedule_message, schedule_time_utc)
+
+        # 1.4 cancel schedule
+        topic_sender.cancel_scheduled_messages(sequence_number)
+        topic_sender.cancel_scheduled_messages(batch_sequence_numbers)
+
+# 2 Receive
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+queue_receiver = sb_client.get_queue_receiver(queue_name="queue_name")
+
+# 2.1 peek
+with sb_client:
+    with queue_receiver:
+        msgs = queue_receiver.peek(count=10)
+        for msg in msgs:
+            print(msg)
+
+# 2.2 iterator receive
+with sb_client:
+    with queue_receiver:
+        for msg in queue_receiver:
+            print(message)
+            msg.complete()
+            # msg.renew_lock()
+            # msg.abondon()
+            # msg.defer()
+            # msg.dead_letter()
+
+# 2.3 pull mode receive
+with sb_client:
+    with queue_receiver:
+        msgs = queue_receiver.receive(max_batch_size=10)
```

Comment:
What comes back is just a plain array, right? I'm wondering if we should migrate away from calling these "batches" since we'll have an actual 'batched' type later.

(same issue in JS btw, we need to figure out our terminology)

Reply:
I think most users would still handle the message one by one.
another layer of abstraction doesn't make their life easier.

Just wondering what benefits we would buy if we introduce another `BatchReceivedMessage` type?